app/controllers/rewards_controller.rb

```diff
@@ -2,7 +2,8 @@ class RewardsController < ApplicationController
   before_action :authenticate_employee!
 
   def index
-    @rewards = Reward.all.page params[:page]
+    @categories = Category.all
+    @rewards = RewardsQuery.new(categories: @categories, rewards: (Reward.all.page params[:page]), params: params).call.includes([:category])
```

Comment:
You can simplify it and make it more readable by passing `RewardsQuery` result to `.page` method.
It would be similar to previous way of calling paging the query result.

Reply:
Yep, and you should paginate results of query, not before - because you can have different number of records on the page when you first paginate and then filter

Reply:
Changed 🙂